espresso-server/app/src/androidTest/java/io/appium/espressoserver/lib/model/MobileBackdoorMethod.java

```diff
@@ -0,0 +1,52 @@
+package io.appium.espressoserver.lib.model;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import javax.annotation.Nullable;
+
+import io.appium.espressoserver.lib.helpers.BackdoorUtils;
+
+@SuppressWarnings("unused")
+public class MobileBackdoorMethod {
+    private String name;
+
+    @Nullable
+    private List<BackdoorMethodArg> args;
+
+    public String getName() {
+        return name;
+    }
+
+    @Nullable
+    public List<BackdoorMethodArg> getRawArgs() {
+        if (args == null) {
+            return Collections.emptyList();
+        }
+        return args;
+    }
+
+    public void setArgs(List<BackdoorMethodArg> args){
+        this.args = args;
+    }
+
+    public Class<?>[] getArgumentTypes() {
+        List<BackdoorMethodArg> rawArgs = getRawArgs();
+        Class<?>[] types = new Class<?>[rawArgs.size()];
+        for (int i = 0; i < rawArgs.size(); i++) {
+            types[i] = BackdoorUtils.parseType(rawArgs.get(i).getType());
+        }
+        return types;
+    }
+
+    public Object[] getArguments() {
+        List<BackdoorMethodArg> rawArgs = getRawArgs();
+        Object[] parsedArgs = new Object[rawArgs.size()];
+        for (int i = 0; i < rawArgs.size(); i++) {
+            parsedArgs[i] = BackdoorUtils.parseValue(rawArgs.get(i).getValue(),
```

Comment:
I think there might be problems here related to how GSON performs deserialization. 
We should consider typefixing for types like int/byte (GSON produces Long by default) and char (String is produced).
It would be also handy to cover it in unit tests as well

Reply:
We are instructing GSON to deserialize all values in String in BackdoorMethodArg (all attributes are String type). Then we parse String values based on their type provided by the user.

Reply:
I know, but it is anyway better to make sure it works as expected by covering it with tests

Reply:
DId not get which part of the code you are referring here to be covered by the test. We do type fixing in BackdoorUtils and there are some tests there to check that. It assumes that GSON has parsed everything into string.

Reply:
We should cover the part where the actual json string is transformed into arrays of argument values and types

Reply:
Ok.